server/src/main/java/org/apache/iotdb/db/service/thrift/impl/TSServiceImpl.java

```diff
@@ -302,9 +320,92 @@ public TSFetchResultsResp call() throws Exception {
 
   protected final ServiceProvider serviceProvider;
 
+  /* Double write module, temporary use */
+  // DOUBLEWRITE GENERAL
+  private final boolean isEnableDoubleWrite;
+  private final boolean isSyncDoubleWrite;
+  private final SessionPool doubleWriteSessionPool;
+  private final DoubleWriteProtectorService doubleWriteProtectorService;
+
+  // DOUBLEWRITE SYNC
+  private final ExecutorService doubleWriteTaskThreadPool;
+
+  // DOUBLEWRITE ASYNC
+  private final DoubleWriteProducer doubleWriteProducer;
+
   public TSServiceImpl() {
     super();
     serviceProvider = IoTDB.serviceProvider;
+
+    IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+    if (config.isEnableDoubleWrite()) {
+      /* Open double write */
+
+      // basic config
+      isEnableDoubleWrite = config.isEnableDoubleWrite();
+      isSyncDoubleWrite = config.isSyncDoubleWrite();
+
+      // create SessionPool for double write
+      doubleWriteSessionPool =
+          new SessionPool(
+              config.getSecondaryAddress(),
+              config.getSecondaryPort(),
+              config.getSecondaryUser(),
+              config.getSecondaryPassword(),
+              5);
+
+      // create DoubleWriteProtectorService
+      Lock firstLogLock = new ReentrantLock();
+      Condition firstLogCondition = firstLogLock.newCondition();
+      doubleWriteProtectorService =
+          new DoubleWriteProtectorService(firstLogLock, firstLogCondition, doubleWriteSessionPool);
+      new Thread(doubleWriteProtectorService).start();
+      try {
+        firstLogLock.lock();
+        firstLogCondition.await();
+        firstLogLock.unlock();
```

Comment:
unlock should be in finally block

Reply:
This unlock statement is only for thread-safely create the firset DoubleWriteLog.